src/google/protobuf/util/internal/default_value_objectwriter.cc

```diff
@@ -165,7 +165,10 @@ DefaultValueObjectWriter* DefaultValueObjectWriter::RenderBytes(
   if (current_ == NULL) {
     ow_->RenderBytes(name, value);
   } else {
-    RenderDataPiece(name, DataPiece(value, false, true));
+    // Since StringPiece is essentially a pointer, takes a copy of "value" to
+    // avoid ownership issues.
+    string_values_.push_back(new string(value.ToString()));
+    RenderDataPiece(name, DataPiece(*string_values_.back(), false, true));
```

Comment:
Does RenderDataPiece cache the StringPiece value somewhere? I thought RenderDataPiece will write directly to output and won't have ownership issues even when the StringPiece is a temporary.